### PR TITLE
[risk=no][RW-4377] Puppeteer new workspace-ui.spec test

### PR DESCRIPTION
**Pull request by aweng98**
A simple workspace-detail UI test that verify elements existence and havior of a checkbox.

### CODE REVIEW

**Comment by jmthibault79** [comment]
Are the screenshot files intended?  If not, please remove them and add a gitignore rule to exclude them

File: e2e/README.md [hunk 1]
```diff
@@ -1,21 +1,18 @@
...
 
-### Try running tests in localhost
-#### clone github project
+#### Clone github project
 - `git clone git@github.com:all-of-us/workbench.git`
 
-#### set up login credential
+#### Set up login credential
 - copy `.env.sample` file. Save new file as `.env` (do not commit `.env` file to version control)
 - edit `.env` to provide workbench login credential
```
Comment (jmthibault79) [inline e2e/README.md]: Add text here describing how it needs to be an existing user in the Test environment without 2FA

File: e2e/package.json [hunk 2]
```diff
@@ -50,18 +55,20 @@
...
+  "dependencies": {
+  },
   "devDependencies": {
+    "@types/faker": "^4.1.9",
     "@types/jest": "^24.0.21",
     "@types/node": "^12.12.5",
-    "@types/puppeteer-core": "^1.9.0",
+    "@types/puppeteer": "^2.0.0",
+    "puppeteer": "^2.1.1",
     "chrome-paths": "^1.0.1",
```
Comment (jmthibault79) [inline e2e/package.json]: can remove this now

File: performance/yarn.lock [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (jmthibault79) [inline performance/yarn.lock]: this file is not related and should be removed

File: e2e/tests/workspace/workspace-ui.spec.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    const diseaseNameCheckbox = await elementDiseaseName.checkbox();
+    expect(diseaseNameCheckbox).toBeTruthy();
+    expect(await getProperty(page, diseaseNameCheckbox, 'checked')).toBeFalsy();
+    expect(await getProperty(page, diseaseNameCheckbox, 'disabled')).toBeFalsy();
+
+    const diseaseNameTextInput = await elementDiseaseName.textfield();
+    expect(diseaseNameTextInput).toBeTruthy();
+    expect(await getProperty(page, diseaseNameTextInput, 'disabled')).toBeTruthy();
+
+    // await (diseaseNameCheckbox).click(); // click checkbox to change
```
Comment (jmthibault79) [inline e2e/tests/workspace/workspace-ui.spec.ts]: how do these 2 lines differ in approach?

File: e2e/tests/workspace/workspace-ui.spec.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    await page.evaluate(elem => elem.click(), diseaseNameCheckbox);
+    // TODO wait async for checked and disabled checking or test will fail
+    await page.waitFor(1000);
+    expect(await getProperty(page, await elementDiseaseName.checkbox(), 'checked')).toBeTruthy();
+    expect(await getProperty(page, await elementDiseaseName.checkbox(), 'disabled')).toBeFalsy();
+
+
+    // check Workspace Name textfield
+    expect(await workspace.input_workspace_name()).toBeTruthy();
```
Comment (jmthibault79) [inline e2e/tests/workspace/workspace-ui.spec.ts]: move these 2 checks above the checkbox checks, since they're first on the page

File: e2e/tests/workspace/workspace-ui.spec.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+    // check dataset select option
+    expect(await workspace.select_dataSet()).toBeTruthy();
+
+    // check all fields in Question #1. What is the primary purpose of your project?
+    expect(await workspace.element_question1_populationHealth().checkbox()).toBeTruthy();
+    expect(await workspace.element_question1_MethodsDevelopment().checkbox()).toBeTruthy();
+    expect(await workspace.element_question1_DrugTherapeuticsDevelopment().checkbox()).toBeTruthy();
+    expect(await workspace.element_question1_ResearchControl().checkbox()).toBeTruthy();
+    expect(await workspace.element_question1_EducationalPurpose().checkbox()).toBeTruthy();
```
Comment (jmthibault79) [inline e2e/tests/workspace/workspace-ui.spec.ts]: there's a For Profit Purpose box in here as well

File: e2e/tests/workspace/workspace-ui.spec.ts [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    expect(await workspace.textarea_question2()).toBeTruthy();
+    expect(await workspace.textarea_question3()).toBeTruthy();
+    expect(await workspace.textarea_question4()).toBeTruthy();
+
+    // Question #5 out-of-box behavior
+    expect(await workspace.radiobutton_question5_notFocusingSpecificPopulation.get()).toBeTruthy();
+    expect(await workspace.radiobutton_question5_notFocusingSpecificPopulation.isChecked()).toBeTruthy();
+
+    // Question: Request a review ... out-of-box behavior
+    expect(await workspace.radiobutton_request_review_yes()).toBeTruthy();
```
Comment (jmthibault79) [inline e2e/tests/workspace/workspace-ui.spec.ts]: Please check that these boxes are disabled and unchecked to start, and that choosing Yes here will enable them.

File: e2e/tests/workspace/workspace-ui.spec.ts [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+
+    // check Workspace Name textfield
+    expect(await workspace.input_workspace_name()).toBeTruthy();
+
+    // check dataset select option
+    expect(await workspace.select_dataSet()).toBeTruthy();
+
+    // check all fields in Question #1. What is the primary purpose of your project?
+    expect(await workspace.element_question1_populationHealth().checkbox()).toBeTruthy();
```
Comment (jmthibault79) [inline e2e/tests/workspace/workspace-ui.spec.ts]: can check that these are unchecked and enabled too

File: e2e/app/Workspaces.ts [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+  public async radiobutton_request_review_no(): Promise<ElementHandle> {
+    return await this.puppeteerPage.waitForXPath(selectors.radio_request_review_no);
+  }
+
+}
+
+export default class Workspaces extends CommonFields {
+
+  public buttonCreateWorkspace = new Button(this.puppeteerPage,'Create Workspace');
+  public radiobutton_question5_notFocusingSpecificPopulation = new Radio(this.puppeteerPage, 'No, I am not interested in focusing on specific population(s) in my research');
```
[the remark this answers is not in the document]
Comment (aweng98) [reply e2e/app/Workspaces.ts]: Done

File: e2e/browser/ChromeBrowser.ts [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+    this.incognito = incognito || true;
+  }
+
+  public async newBrowser(opts?): Promise<Puppeteer.Browser> {
+    const launchOptions = opts || this.launchOpts;
+    return this.browser = await Puppeteer.launch(launchOptions);
+  }
+
+  public async newPage(browser?: Puppeteer.Browser): Promise<Puppeteer.Page> {
+    console.log("isHeadless: " + configs.isHeadless);
```
[the remark this answers is not in the document]
Comment (aweng98) [reply e2e/browser/ChromeBrowser.ts]: removed

File: e2e/README.md (hunk 1 above)
Comment (aweng98) [reply e2e/README.md]: done

File: e2e/package.json (hunk 2 above)
Comment (aweng98) [reply e2e/package.json]: done

File: performance/yarn.lock (hunk 3 above)
Comment (aweng98) [reply performance/yarn.lock]: done

File: e2e/tests/workspace/workspace-ui.spec.ts [hunk 11]
```diff
@@ -0,0 +1,105 @@
+import Workspaces from '../../app/Workspaces';
```
[the remark this answers is not in the document]
Comment (aweng98) [reply e2e/tests/workspace/workspace-ui.spec.ts]: they are different pages. tests are grouped by pages at a higher level.

File: e2e/tests/workspace/workspace-ui.spec.ts (hunk 4 above)
Comment (aweng98) [reply e2e/tests/workspace/workspace-ui.spec.ts]: commented Puppeteer `.click()` function failed to click checkbox. It could be an issue in Puppeteer tool. But I don't really understand why. `click` by JS native function worked.

File: e2e/tests/workspace/workspace-ui.spec.ts (hunk 5 above)
Comment (aweng98) [reply e2e/tests/workspace/workspace-ui.spec.ts]: done

File: e2e/tests/workspace/workspace-ui.spec.ts (hunk 6 above)
Comment (aweng98) [reply e2e/tests/workspace/workspace-ui.spec.ts]: done

**Comment by aweng98** [comment]
@jmthibault79 Latest changes are in. thanks
